src/framework/ui/tests/navigationcontroller_tests.cpp

```diff
@@ -0,0 +1,295 @@
+
+            Control* c = make_control(idx);
+
+            p->controls.push_back(c);
+            p->icontrols.insert(c->control);
+        }
+
+        p->panel = new NiceMock<NavigationPanelMock>();
+        ON_CALL(*p->panel, enabled()).WillByDefault(Return(true));
+        ON_CALL(*p->panel, active()).WillByDefault(Return(false));
+        ON_CALL(*p->panel, forceActiveRequested()).WillByDefault(Return(async::Channel<PanelControl>()));
+        ON_CALL(*p->panel, controls()).WillByDefault(ReturnRef(p->icontrols));
+
+        INavigation::Index& idx = make_idx(env);
+        idx.setOrder(panelOrder);
+        ON_CALL(*p->panel, index()).WillByDefault(ReturnRef(idx));
+
+        return p;
+    }
+
+    Section* make_section(Env& env, int sectOrder, size_t panelsCount, size_t controlsCount)
+    {
+        Section* s = new Section();
+
+        for (size_t pi = 0; pi < panelsCount; ++pi) {
+            Panel* p = make_panel(env, pi, controlsCount);
+            s->panels.push_back(p);
+            s->ipanels.insert(p->panel);
+        }
+
+        s->section = new NiceMock<NavigationSectionMock>();
+        ON_CALL(*s->section, enabled()).WillByDefault(Return(true));
+        ON_CALL(*s->section, active()).WillByDefault(Return(false));
+        ON_CALL(*s->section, forceActiveRequested()).WillByDefault(Return(async::Channel<SectionPanelControl>()));
+        ON_CALL(*s->section, panels()).WillByDefault(ReturnRef(s->ipanels));
+
+        INavigation::Index& idx = make_idx(env);
+        idx.setOrder(sectOrder);
+        ON_CALL(*s->section, index()).WillByDefault(ReturnRef(idx));
+
+        return s;
+    }
+
+    void print(Section* s)
+    {
+        LOGI() << "section: " << s->section->name() << ", idx: " << s->section->index().to_string()
+               << ", active: " << s->section->active() << ", enabled: " << s->section->enabled();
+
+        for (const Panel* p : s->panels) {
+            LOGI() << "panel: " << p->panel->name() << ", idx: " << p->panel->index().to_string()
+                   << ", active: " << p->panel->active() << ", enabled: " << p->panel->enabled();
+
+            for (const Control* c : p->controls) {
+                LOGI() << "control: " << c->control->name() << ", idx: " << c->control->index().to_string()
+                       << ", active: " << c->control->active() << ", enabled: " << c->control->enabled();
+            }
+        }
+    }
+};
+
+TEST_F(NavigationControllerTests, FirstActiveOnNextSection)
+{
+    Env env;
+
+    //! CASE Nothing active, and we call next section (F6)
+
+    //! GIVEN Two section, not active
+    Section* sect1 = make_section(env, 1, 2, 3);
+    Section* sect2 = make_section(env, 2, 2, 3);
+
+    env.controller.reg(sect1->section);
+    env.controller.reg(sect2->section);
+
+    //! CHECK The first section, the first panel, the first control must be activated.
+    EXPECT_CALL(*sect1->section, setActive(true));
+    EXPECT_CALL(*sect1->panels[0]->panel, setActive(true));
```

Comment:
I think it is better to add a check for panels/controls size:

ASSERT_FALSE(sect1->panels.isEmpty())
ASSERT_FALSE(sect1->panels[0].controls.isEmpty())

Reply:
> I think it is better to add a check for panels/controls size:
> 
> ASSERT_FALSE(sect1->panels.isEmpty())
> ASSERT_FALSE(sect1->panels[0].controls.isEmpty())

I think this is redundant because these panels and sections are created not by the code under test, but by the test itself. There are no conditions when lists can be empty, unless, of course, you explicitly set the count to zero, but in this case, I hope the developer understands what he is doing :)